src/components/vector/ExportDialog.tsx

```
@@ -53,44 +56,92 @@ export function ExportDialog({ open, onOpenChange }: ExportDialogProps) {
+      // Manejar las diferentes formas de vectores
       switch(shape) {
         case 'line':
           // Línea simple
-          return `<line id="${id}" x1="${-halfLen}" y1="0" x2="${halfLen}" y2="0" ${stroke} ${transform} />`;
+          return `<line id="${id}" x1="0" y1="0" x2="${actualLength}" y2="0" ${stroke} ${transform} />`;
         
         case 'arrow':
           // Flecha - línea con punta de flecha
-          const arrowSize = settings.vectorLength * 0.25;
+          const arrowSize = actualLength * 0.25;
           return `<g id="${id}" ${transform}>
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)